core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/PatternPropertyShape.java

```diff
@@ -36,6 +37,7 @@
 
 		this.pattern = pattern;
 		this.flags = flags;
+		this.Expected = SimpleValueFactory.getInstance().createLiteral("Flags: " + flags + " : Pattern: " + pattern);
```

Comment:
Should be pattern first. Also need to handle when flags is empty or null (I can't remember if flags can be null at this point).